client/blocks/payment-request/payment-request-express.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import { Elements, PaymentRequestButtonElement } from '@stripe/react-stripe-js';
+
+/**
+ * Internal dependencies
+ */
+import { getStripeServerData } from '../stripe-utils';
+import { useInitialization } from './use-initialization';
+import { useCheckoutSubscriptions } from './use-checkout-subscriptions';
+import { ThreeDSecurePaymentHandler } from '../three-d-secure';
+import { GooglePayButton, shouldUseGooglePayBrand } from './branded-buttons';
+import { CustomButton } from './custom-button';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+/**
+ * PaymentRequestExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const PaymentRequestExpressComponent = ( {
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	setExpressPaymentError,
+	emitResponse,
+	onClick,
+	onClose,
+} ) => {
+	const {
+		paymentRequest,
+		paymentRequestEventHandlers,
+		clearPaymentRequestEventHandler,
+		isProcessing,
+		canMakePayment,
+		onButtonClick,
+		abortPayment,
+		completePayment,
+		paymentRequestType,
+	} = useInitialization( {
+		billing,
+		shippingData,
+		setExpressPaymentError,
+		onClick,
+		onClose,
+		onSubmit,
+	} );
+	useCheckoutSubscriptions( {
+		canMakePayment,
+		isProcessing,
+		eventRegistration,
+		paymentRequestEventHandlers,
+		clearPaymentRequestEventHandler,
+		billing,
+		shippingData,
+		emitResponse,
+		paymentRequestType,
+		completePayment,
+		abortPayment,
+	} );
+
+	// locale is not a valid value for the paymentRequestButton style.
+	// Make sure `theme` defaults to 'dark' if it's not found in the server provided configuration.
+	const { theme = 'dark' } = getStripeServerData().button;
+
+	const paymentRequestButtonStyle = {
+		paymentRequestButton: {
+			type: 'default',
```

Comment:
@reykjalin note we need to use `wc_stripe_payment_request_params.button.type` here, so the label is changed to "Donate" or "Buy" when it's set.

Reply:
I think we should make this part of #1524 as part of "using the same APIs", but this is an excellent callout.

Essentially; we shouldn't be registering 2 different settings objects and I think fixing that fits well within the scope of #1524

Reply:
Oops, sorry, this will be fixed with #1516! I forgot I opened an issue for this specifically 🙃

Reply:
@reykjalin or you could use something like this in this PR, so the height and type props are applied to the button.

```JavaScript
const {
	type = 'default',
	theme = 'dark',
	height = '48',
} = getStripeServerData().button;

const paymentRequestButtonStyle = {
	paymentRequestButton: {
		type,
		theme,
		height: height + 'px',
	}
}
```

Reply:
Ah yeah, that's a good idea.

Fixed in 0e57169b0a0d848cf833e46a75611668e7f9ff61 🙌